### PR TITLE
Implement Path-GASP

### PR DESCRIPTION
**Pull request by ketphan02**
### Issue
- We want to benchmark the algorithms from Jeff's thesis
- For now, I choose one of the most naive algorithm: **Path-GASP**

### Changes
- Added new algorithm under `benchmarking/other_algorithms/path_gasp.py`

### Graph
image
Note: this graph only run for class size = 8 and 12

### CODE REVIEW

**Review by Opeyem1a — COMMENTED** [review]
I think we should meet at some point to discuss this, I kinda dislike this approach on an architectural level and there is a misconception or two about some of the existing stuff (I think)

**Review by Opeyem1a — CHANGES_REQUESTED** [review]
For now though, I don't think we should merge this as is

File: benchmarking/data/simulated_data/mock_student_provider.py [hunk 1]
```diff
@@ -90,11 +90,21 @@ def create_mock_students(
...
+        # Generate preference for student
+        preferences = random.sample(projects, max_num_preferences)
+        random.shuffle(preferences)
+
         students.append(
             Student(
                 i,
                 relationships=relationships,
                 attributes=attributes,
+                preferences=preferences,
```
[the remark this answers is not in the document]
Comment (ketphan02) [reply benchmarking/data/simulated_data/mock_student_provider.py:107]: You're right! Then should we remove preferences field in Student?

**Comment by ketphan02** [comment]
We do not need Path-GASP since it's too slow and too naive

**Pull request closed without merging**